### ISSUE TITLE
Tests fail with PHP 7.1

### ISSUE THREAD

**Issue by siwinski** [njh/easyrdf#276]
```
$ make test-lib/usr/bin/php71 composer.phar --no-ansi --no-interaction install
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 44 installs, 0 updates, 0 removals
  - Installing symfony/process (v2.8.14)
    Loading from cache

  - Installing symfony/yaml (v2.8.14)
    Loading from cache

  - Installing symfony/filesystem (v2.8.14)
    Loading from cache

  - Installing symfony/finder (v2.8.14)
    Loading from cache

  - Installing symfony/polyfill-mbstring (v1.3.0)
    Loading from cache

  - Installing psr/log (1.0.2)
    Loading from cache

  - Installing symfony/debug (v3.0.9)
    Loading from cache

  - Installing symfony/console (v2.8.14)
    Loading from cache

  - Installing michelf/php-markdown (1.7.0)
    Loading from cache

  - Installing nikic/php-parser (v0.9.5)
    Loading from cache

  - Installing twig/twig (v1.28.2)
    Loading from cache

  - Installing pimple/pimple (v2.1.1)
    Downloading

  - Installing sami/sami (v2.0.0)
    Downloading

  - Installing sebastian/version (1.0.6)
    Loading from cache

  - Installing sebastian/global-state (1.1.1)
    Loading from cache

  - Installing sebastian/recursion-context (1.0.2)
    Loading from cache

  - Installing sebastian/exporter (1.2.2)
    Loading from cache

  - Installing sebastian/environment (1.3.8)
    Loading from cache

  - Installing sebastian/diff (1.4.1)
    Loading from cache

  - Installing sebastian/comparator (1.2.2)
    Loading from cache

  - Installing doctrine/instantiator (1.0.5)
    Loading from cache

  - Installing phpunit/php-text-template (1.2.1)
    Loading from cache

  - Installing phpunit/phpunit-mock-objects (2.3.8)
    Loading from cache

  - Installing phpunit/php-timer (1.0.8)
    Loading from cache

  - Installing phpunit/php-file-iterator (1.4.2)
    Loading from cache

  - Installing phpunit/php-token-stream (1.4.9)
    Loading from cache

  - Installing phpunit/php-code-coverage (2.2.4)
    Loading from cache

  - Installing webmozart/assert (1.2.0)
    Downloading

  - Installing phpdocumentor/reflection-common (1.0)
    Loading from cache

  - Installing phpdocumentor/type-resolver (0.2.1)
    Downloading

  - Installing phpdocumentor/reflection-docblock (3.1.1)
    Loading from cache

  - Installing phpspec/prophecy (v1.6.2)
    Loading from cache

  - Installing phpunit/phpunit (4.8.29)
    Loading from cache

  - Installing semsol/arc2 (2.3.0)
    Downloading

  - Installing squizlabs/php_codesniffer (1.4.8)
    Loading from cache

  - Installing zendframework/zend-stdlib (3.1.0)
    Downloading

  - Installing container-interop/container-interop (1.1.0)
    Loading from cache

  - Installing zendframework/zend-validator (2.8.1)
    Downloading

  - Installing zendframework/zend-escaper (2.5.2)
    Downloading

  - Installing zendframework/zend-uri (2.5.2)
    Downloading

  - Installing zendframework/zend-loader (2.5.1)
    Downloading

  - Installing zendframework/zend-http (2.5.5)
    Downloading

  - Installing ml/iri (1.1.4)
    Downloading

  - Installing ml/json-ld (1.0.7)
    Downloading

symfony/console suggests installing symfony/event-dispatcher ()
sebastian/global-state suggests installing ext-uopz (*)
phpunit/phpunit-mock-objects suggests installing ext-soap (*)
phpunit/php-code-coverage suggests installing ext-xdebug (>=2.2.1)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
zendframework/zend-validator suggests installing zendframework/zend-db (Zend\Db component)
zendframework/zend-validator suggests installing zendframework/zend-filter (Zend\Filter component, required by the Digits validator)
zendframework/zend-validator suggests installing zendframework/zend-i18n (Zend\I18n component to allow translation of validation error messages as well as to use the various Date validators)
zendframework/zend-validator suggests installing zendframework/zend-i18n-resources (Translations of validator messages)
zendframework/zend-validator suggests installing zendframework/zend-math (Zend\Math component)
zendframework/zend-validator suggests installing zendframework/zend-servicemanager (Zend\ServiceManager component to allow using the ValidatorPluginManager and validator chains)
zendframework/zend-validator suggests installing zendframework/zend-session (Zend\Session component)
Writing lock file
Generating autoload files
mkdir -p reports
/usr/bin/php71 vendor/bin/phpunit  -c config/phpunit.xml --testsuite "EasyRdf Library"
PHPUnit 4.8.29 by Sebastian Bergmann and contributors.
Warning:	Deprecated configuration setting "strict" used

.............................................................   61 / 1343 (  4%)
.............................................................  122 / 1343 (  9%)
.............................................................  183 / 1343 ( 13%)
.............................................................  244 / 1343 ( 18%)
.............................................................  305 / 1343 ( 22%)
.............................................................  366 / 1343 ( 27%)
...........................................................I.  427 / 1343 ( 31%)
.............................................................  488 / 1343 ( 36%)
.............................................................  549 / 1343 ( 40%)
.............................................................  610 / 1343 ( 45%)
.............................................................  671 / 1343 ( 49%)
.............................................................  732 / 1343 ( 54%)
....I.....................................................I..  793 / 1343 ( 59%)
..............................................I......II......  854 / 1343 ( 63%)
.........................................I.....IIIIIII.......  915 / 1343 ( 68%)
SS...................................................I.......  976 / 1343 ( 72%)
............................................................. 1037 / 1343 ( 77%)
............................................................. 1098 / 1343 ( 81%)
....................EEEEEEF.................................. 1159 / 1343 ( 86%)
............................................................. 1220 / 1343 ( 90%)
............................................................. 1281 / 1343 ( 95%)
............................................................. 1342 / 1343 ( 99%)
.

Time: 661 ms, Memory: 20.00MB

There were 6 errors:

1) EasyRdf\Serialiser\GraphVizTest::testSerialiseDot
Error: [] operator not supported for strings

/tmp/easyrdf/lib/Serialiser/GraphViz.php:228
/tmp/easyrdf/lib/Serialiser/GraphViz.php:245
/tmp/easyrdf/lib/Serialiser/GraphViz.php:288
/tmp/easyrdf/lib/Serialiser/GraphViz.php:386
/tmp/easyrdf/test/EasyRdf/Serialiser/GraphVizTest.php:124

2) EasyRdf\Serialiser\GraphVizTest::testSerialiseDotUseLabels
Error: [] operator not supported for strings

/tmp/easyrdf/lib/Serialiser/GraphViz.php:228
/tmp/easyrdf/lib/Serialiser/GraphViz.php:245
/tmp/easyrdf/lib/Serialiser/GraphViz.php:288
/tmp/easyrdf/lib/Serialiser/GraphViz.php:386
/tmp/easyrdf/test/EasyRdf/Serialiser/GraphVizTest.php:152

3) EasyRdf\Serialiser\GraphVizTest::testSerialiseDotOnlyLabelled
Error: [] operator not supported for strings

/tmp/easyrdf/lib/Serialiser/GraphViz.php:228
/tmp/easyrdf/lib/Serialiser/GraphViz.php:245
/tmp/easyrdf/lib/Serialiser/GraphViz.php:288
/tmp/easyrdf/lib/Serialiser/GraphViz.php:386
/tmp/easyrdf/test/EasyRdf/Serialiser/GraphVizTest.php:182

4) EasyRdf\Serialiser\GraphVizTest::testSerialisePng
Error: [] operator not supported for strings

/tmp/easyrdf/lib/Serialiser/GraphViz.php:228
/tmp/easyrdf/lib/Serialiser/GraphViz.php:245
/tmp/easyrdf/lib/Serialiser/GraphViz.php:288
/tmp/easyrdf/lib/Serialiser/GraphViz.php:358
/tmp/easyrdf/lib/Serialiser/GraphViz.php:390
/tmp/easyrdf/test/EasyRdf/Serialiser/GraphVizTest.php:207

5) EasyRdf\Serialiser\GraphVizTest::testSerialiseGif
Error: [] operator not supported for strings

/tmp/easyrdf/lib/Serialiser/GraphViz.php:228
/tmp/easyrdf/lib/Serialiser/GraphViz.php:245
/tmp/easyrdf/lib/Serialiser/GraphViz.php:288
/tmp/easyrdf/lib/Serialiser/GraphViz.php:358
/tmp/easyrdf/lib/Serialiser/GraphViz.php:390
/tmp/easyrdf/test/EasyRdf/Serialiser/GraphVizTest.php:223

6) EasyRdf\Serialiser\GraphVizTest::testSerialiseSvg
Error: [] operator not supported for strings

/tmp/easyrdf/lib/Serialiser/GraphViz.php:228
/tmp/easyrdf/lib/Serialiser/GraphViz.php:245
/tmp/easyrdf/lib/Serialiser/GraphViz.php:288
/tmp/easyrdf/lib/Serialiser/GraphViz.php:358
/tmp/easyrdf/lib/Serialiser/GraphViz.php:390
/tmp/easyrdf/test/EasyRdf/Serialiser/GraphVizTest.php:239

--

There was 1 failure:

1) EasyRdf\Serialiser\GraphVizTest::testDotNotFound
Failed asserting that exception of type "Error" matches expected exception "EasyRdf\Exception".

FAILURES!
Tests: 1343, Assertions: 2660, Errors: 6, Failures: 1, Skipped: 2, Incomplete: 15.
Makefile:59: recipe for target 'test-lib' failed
make: *** [test-lib] Error 2
```

**Comment by siwinski** [comment]
https://bugs.php.net/bug.php?id=72430

**Comment by siwinski** [comment]
http://php.net/manual/en/migration71.incompatible.php#migration71.incompatible.empty-string-index-operator

**Comment by njh** [comment]
Thanks for reporting.

**Comment by njh** [comment]
PHP 7.1 was added to Travis in 5d911642aeacb3fe1d64d4f3e5ab35dddbe180af

This bug was fixed in 4ed264051ed407d59c2fde4128be176e96e8f22a
